src/compat/db_query.c

```diff
@@ -600,6 +633,38 @@ db_unpack_prepare_info (DB_PREPARE_INFO * info, char *buffer)
 	  ptr = or_unpack_domain (ptr, &info->host_var_expected_domains[i], NULL);
 	}
     }
+
+  /* cte */
+  ptr = or_unpack_int (ptr, &info->cte_num_query);
+  cte_num = info->cte_num_query;
+
+  if (cte_num > 0)
+    {
+      info->cte_info = (DB_PREPARE_CTE_INFO *) malloc (cte_num * sizeof (DB_PREPARE_CTE_INFO));
+      if (info->cte_info == NULL)
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1,
+		  cte_num * sizeof (DB_PREPARE_CTE_INFO));
+	  goto error;
+	}
+      for (q = 0; q < cte_num; q++)
+	{
+	  OR_UNPACK_XASL_ID (ptr, &info->cte_info[q].cte_xasl_id);
+	  ptr = or_unpack_int (ptr, &info->cte_info[q].cte_host_var_count);
+	  info->cte_info[q].cte_host_var_index = (int *) malloc (info->cte_info[q].cte_host_var_count * sizeof (int));
```

Comment:
Isn't there any possibility that cte_host_var_count = 0?

Reply:
good point. thanks